Model/Checkout/PlaceOrder.php

```diff
@@ -238,15 +238,17 @@ public function createOrder(Quote $quote, string $amwalOrderId, string $refId):
         $this->logDebug(sprintf('Submitting quote with ID %s', $quote->getId()));
         $order = $this->getOrderByAmwalOrderId($amwalOrderId);
 
-        if ($order->getEntityId()) {
-            if ($order->getState() == Order::STATE_PENDING_PAYMENT) {
-                $this->orderRepository->delete($order);
-                $order = $this->quoteManagement->submit($quote);
-            }
-        }else{
-            $order = $this->quoteManagement->submit($quote);
+        if ($order) {
```

Comment:
we need to add additional checking that this order is pending to make sure we don't cancel a processing order